lib/kube/proxy/forwarder.go

```diff
@@ -520,7 +525,7 @@ type handlerWithAuthFuncStd func(ctx *authContext, w http.ResponseWriter, r *htt
 const accessDeniedMsg = "[00] access denied"
 
 // authenticate function authenticates request
-func (f *Forwarder) authenticate(req *http.Request) (*authContext, error) {
+func (f *Forwarder) authenticate(req *http.Request, params httprouter.Params, routeSourcer routeSourcer) (*authContext, error) {
```

Comment:
I don't fully understand why you need to pass the params/route sourcer here.

Given that `singleCertHandler` already defines the
 ```
o.Identity.RouteToCluster = teleportCluster
o.Identity.KubernetesCluster = kubeCluster
```

Do we ned to have a special case for route sourcer? We can always retrieve it from identity

Reply:
That's a good call, it was left over from before I had it modifying (and relying on) the identity directly. I've removed the `routeSource` mechanism outright and reverted most of my changes to the forwarder itself. I ended up diving deeper into how the identity is passed / trusted between clusters and wrote some notes in a comment.